### PR TITLE
Add pydantic for co2eq parameters

### PR DESCRIPTION
**Pull request by pierresegonne**
## Description

Add model for schema of co2eq parameters with pydantic. 

Verification of the config is executed at runtime when pydantic parses the file to fit the provided model.

This PR only aims at generating a model that loads the co2eq parameters object. 

It does reveal that we could be much smarter about the formatting of that file, and with pydantic it will be easy to verify that any update to the config matches our schema expectations.

### CODE REVIEW

**Comment by pierresegonne** [comment]
Theoretically we could remove `test_co2eq_parameters.py` as well, but I plan on doing this in a subsequent PR.

**Review by madsnedergaard — APPROVED** [review]
This is great! 🥳 
And yeah, when looking at the schema it seems pretty complex - but way improve that after this PR as you said :)
(Approving in case you don't want to wait until Felix is back next week)

File: electricitymap/contrib/config/model.py [hunk 1]
```diff
@@ -84,34 +104,197 @@ class Zone(StrictBaseModel):
...
     parsers: Optional[ExchangeParsers]
     rotation: Optional[int]
 
-    class Config:
-        # To allow for both comment and _comment.
-        # TODO: stick to one of them
-        allow_population_by_field_name = True
+
+class PowerOriginRatiosValues(StrictBaseModelWithAlias):
+    battery_charge: Optional[confloat(ge=0, le=1)] = Field(None, alias="battery charge")
```
[the remark this answers is not in the document]
Comment (pierresegonne) [reply electricitymap/contrib/config/model.py:122]: Hmm good idea, let me check if that works

**Comment by pierresegonne** [comment]
For Felix it's more of a FYI, he's only back in two weeks, and will have a lot to catch up on :)

**Pull request merged**